In [2]:
import pandas as pd

In [4]:
data=pd.read_csv(r'C:\Users\Admin\Downloads\Advertising.csv')
data.head()

Unnamed: 0     TV  Radio  Newspaper  Sales
0           1  230.1   37.8       69.2   22.1
1           2   44.5   39.3       45.1   10.4
2           3   17.2   45.9       69.3    9.3
3           4  151.5   41.3       58.5   18.5
4           5  180.8   10.8       58.4   12.9

# Explain the dataset.

In [5]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 200 entries, 0 to 199
Data columns (total 5 columns):
 #   Column      Non-Null Count  Dtype  
---  ------      --------------  -----  
 0   Unnamed: 0  200 non-null    int64  
 1   TV          200 non-null    float64
 2   Radio       200 non-null    float64
 3   Newspaper   200 non-null    float64
 4   Sales       200 non-null    float64
dtypes: float64(4), int64(1)
memory usage: 7.9 KB


In [6]:
import numpy as np

In [7]:
data.shape

(200, 5)

In [8]:
data.size

1000

In [9]:
data.ndim

2

In [10]:
data.isnull().sum()

Unnamed: 0    0
TV            0
Radio         0
Newspaper     0
Sales         0
dtype: int64

From the data set we have the following observations:
*the dataset contain 200 rows and 5 columns.
*the size is 1000 and dimension is 2.
*there is no null values present in the dataset.
*there are three input variable(ie TV,Radio,Newspaper) and one output variable(ie sales).
*data type is float64.
*no null values present in columns.

# Check the summary statistics and discuss the max,min,avg,median and percentailes.

In [11]:
data.describe()

Unnamed: 0          TV       Radio   Newspaper       Sales
count  200.000000  200.000000  200.000000  200.000000  200.000000
mean   100.500000  147.042500   23.264000   30.554000   14.022500
std     57.879185   85.854236   14.846809   21.778621    5.217457
min      1.000000    0.700000    0.000000    0.300000    1.600000
25%     50.750000   74.375000    9.975000   12.750000   10.375000
50%    100.500000  149.750000   22.900000   25.750000   12.900000
75%    150.250000  218.825000   36.525000   45.100000   17.400000
max    200.000000  296.400000   49.600000  114.000000   27.000000

*the mean value of TV is 147.04250,Radio is 23.2640,Newspaper is 30.5540 and Sales is 14.02250.
*standard deviations are 85.854236,14.846809,21.778621,5.217457.
*the minimum values are 0.70,0,0.30,1.60.
*the maximum values are 296.40,49.60,114.0,27.00.
*the 25% values are 74.3750,9.9750,12.750,10.3750
*the 50% values are 149.750,22.90,25.750,12.90.
*the 75% values are 218.8250,36.5250,45.10,17.40.
Here the minimum value of radio is 0,so we can check is there any other zero value present in the dataset.

In [12]:
(data == 0).sum(axis=0)

Unnamed: 0    0
TV            0
Radio         1
Newspaper     0
Sales         0
dtype: int64

Therefore there is no other zero values present in the dataset.

# The manager wants to find out whether the same amount was spent for the three advertisements(TV,Radio,Newspaper).Coment on your findings.

In [13]:
from scipy import stats

H0:same amount was spent for the three advertisements.
H1:different amount was spent for the advertisements.    

In [14]:
df=pd.DataFrame(data)

In [28]:
A=df['TV']
A

0      230.1
1       44.5
2       17.2
3      151.5
4      180.8
       ...  
195     38.2
196     94.2
197    177.0
198    283.6
199    232.1
Name: TV, Length: 200, dtype: float64

In [29]:
B=df['Radio']
B

0      37.8
1      39.3
2      45.9
3      41.3
4      10.8
       ... 
195     3.7
196     4.9
197     9.3
198    42.0
199     8.6
Name: Radio, Length: 200, dtype: float64

In [30]:
C=df['Newspaper']
C

0      69.2
1      45.1
2      69.3
3      58.5
4      58.4
       ... 
195    13.8
196     8.1
197     6.4
198    66.2
199     8.7
Name: Newspaper, Length: 200, dtype: float64

In [31]:
fvalue,pvalue=stats.f_oneway(A,B,C)
print(fvalue,pvalue)

358.8514595342597 4.552931539744962e-103


In [32]:
alpha=0.05

In [34]:
if(pvalue<=alpha):
    print("Reject the null hypothesis")
else:
        print("Fail to reject the null hypothesis")

Reject the null hypothesis


Since the pvalue is less than alpha value we can reject the null hypothesis.Therefore different amount was spent for three advertisements.